src/lib.rs

```diff
@@ -0,0 +1,325 @@
+    display_name: String,
+    #[serde(rename = "standardAccount")]
+    standard_account: Option<String>,
+    website: Option<String>,
+    symbols: Option<Vec<String>>,
+    decimals: Option<Vec<u8>>,
+}
+
+impl AccountType {
+    fn name(&self) -> String {
+        let mut name = self.network.clone().unwrap_or(
+            if let Some(standard_account) = &self.standard_account {
+                format!("Bare{}", standard_account)
+            } else {
+                let name = self
+                    .network
+                    .as_ref()
+                    .expect("network should not be empty if no account specified")
+                    .to_string();
+
+                assert!(
+                    name.starts_with("reserved"),
+                    "If no account specified, network should start `reserved` not:{}",
+                    name
+                );
+                name
+            },
+        );
+        let is_reserved = name.starts_with("reserved");
+        if name.ends_with("net") {
+            name.truncate(name.len() - 3);
+        }
+        // Capitalise
+        name.get_mut(0..1)
+            .expect("name should not be empty")
+            .make_ascii_uppercase();
+
+        let postfix = if is_reserved { "" } else { "Account" };
+        format!("{}{}", rust_valid_id(name), postfix)
+    }
+}
+
+fn rust_valid_id(name: String) -> String {
+    // TODO find function that already does this. `-` are excluded in particular.
+    name.chars().filter(|c| c.is_alphanumeric()).collect()
+}
+
+/// Creates the Ss58AddressFormat enum from the ss58-registry.json file
+#[proc_macro]
+pub fn ss58_registry(input: TokenStream) -> TokenStream {
+    assert!(input.is_empty(), "No arguments are expected");
+
+    match create_ss58_registry(include_str!("ss58-registry.json")) {
+        Ok(result) => result,
+        Err(msg) => panic!("{}", msg),
+    }
+}
+
+fn create_ss58_registry(json: &str) -> Result<TokenStream, String> {
+    let registry: Registry = serde_json::from_str(json).expect("valid json file");
+
+    registry.is_valid()?;
+
+    let identifier: Vec<_> = registry
+        .registry
+        .iter()
+        .map(|r| format_ident!("{}", r.name()))
+        .collect();
+
+    let reserved_identifiers: Vec<_> = registry
+        .registry
+        .iter()
+        .filter(|r| r.network.as_ref().filter(|r|r.starts_with("reserved")).is_some())
+        .map(|r| format_ident!("{}", r.name()))
+        .collect();
+
+    let reserved_numbers: Vec<_> = registry
+        .registry
+        .iter()
+        .filter(|r| r.network.as_ref().filter(|r|r.starts_with("reserved")).is_some())
+        .map(|r| r.prefix)
+        .collect();
+
+    let number: Vec<_> = registry.registry.iter().map(|r| r.prefix).collect();
+    let count = registry.registry.len();
+    let name: Vec<_> = registry
+        .registry
+        .iter()
+        .map(|r| r.network.clone().unwrap_or_else(|| "Bare".to_string()))
```

Comment:
Bare?

Reply:
There are three enum varients which don't have a network associated with them: BareSr25519, BareEd25519 and BareSecp256k1
Possibly there should be an is_bare() or has_network() function to identify this subset?

Reply:
Then we should set it to something in the json and make sure that this can not be null.

Reply:
fixed. Have set network to be BareEd25519 etc. for those ones.